src/components/layout/header/menu-flyout.tsx

```diff
@@ -40,18 +54,25 @@ interface NavigationFlyoutProp {
   onClick?: () => void;
   translation?: string;
   showLanguageSwitch?: boolean;
+  setIsNavigationVisible: (b: boolean) => void;
 }
 
 export const NavigationFlyout: React.FC<NavigationFlyoutProp> = (props) => {
   return (
-    <FullscreenOverlay visible={props.visible} onClick={props.onClick}>
-      <ScrollContainer>
-        <FullHeightNavigation
-          translation={props.translation}
-          showLanguageSwitch={props.showLanguageSwitch}
-        />
-      </ScrollContainer>
-      {props.visible && <NoScrollBody />}
-    </FullscreenOverlay>
+    <>
+      <FullscreenOverlay visible={props.visible} onClick={props.onClick}>
+        <ScrollContainer>
+          <FullHeightNavigation
+            translation={props.translation}
+            showLanguageSwitch={props.showLanguageSwitch}
+          />
+        </ScrollContainer>
+        {props.visible && <NoScrollBody />}
+      </FullscreenOverlay>
+      <BackgroundOverlay
+        visible={props.visible}
+        onClick={() => props.setIsNavigationVisible(false)}
```

Comment:
Ad-hoc request: While working on that, maybe you can provide also the `ESC` key to close it for some additional convenience?

Reply:
Good Idea, I will add it 👍

Reply:
I see the added feature, let's discuss in the appropriate place. Thanks!